### ISSUE TITLE
Should I retrain the pre-trained model in cross-validation on S3DIS Dataset?

### ISSUE THREAD

**Issue by MaxMece** [laughtervv/SGPN#27]
when i use the provided pre-trained model, train on area 1-6 except 5 and test on area 5, i get the admirable result and mAP is about 0.6. 

but when i train on area 2-6 and test on area 1, mAP is about 0.4 and then test on area 2, mAP is about 0.1.

i try many  versions of mingroupsize.txt and pergroup_thres.txt, but 0.1 is still the best result.

so i have the question about the pre-trained model, maybe i shoule retrain the pre-trained model in each cross-validation?

Thanks for your response.

ps:
after training 200 epochs，loss is about 300-400, and grouperr: 0.02 same: 0.01 diff: 0.001 pos: 0.004

**Comment by Xieshichen** [comment]
@ MaxMece, have you encountered the situation that when you run the valid.py there are too much nan in the results?

**Comment by MaxMece** [comment]
我猜你是中国人，所以直接给你回中文了，如果你看不懂，那我再凭我的渣渣英语给你翻过去～

是有碰到很多NaN, 但是我认为NaN是没错的，你仔细看valid.py的代码，它统计per_threshold时，是统计每个类下proposal的ap达到0.8时的阈值，最后取平均，就得到了每个类的per_threshold，但是当一个block中不存在这个类会出现什么情况呢？就会进行除0操作，所以会出现NaN。

valid.py在我看来主要是一个统计工作，具体还得作者指正！

祝好！
孙飞

At 2019-07-01 10:25:50, "Xieshichen" <notifications@github.com> wrote:


@ MaxMece, have you encountered the situation that when you run the valid.py there are too much nan in the results?

**Comment by Xieshichen** [comment]
@ MaxMece，十分感谢你的回复，如果nan在valid.py产生是正常的，那么你生成的per_threshold是否可用（我出现了11个类别的阈值为0情况）？

**Comment by MaxMece** [comment]
可以使用，但是在论文中，作者说valid.py的数据集采用训练集的子集，具体是什么子集并没有指出，我暂时是用全集来用的，但总的来说有时候效果差强人意，使用不同的per_threshold文件test的结果不同，这一块也还在实验中，并且我自己跑出来的per_threshold文件和作者提供的per_threshold文件差别挺大的，作者的每个类的阈值都在10以下，我的基本在10以上或左右。

我怀疑你跑出来0是因为训练部分的原因，最不济也应该是默认的0.2，具体还得你自己多分析分析。

在 2019-07-01 17:50:45，"ShiChenXie" <notifications@github.com> 写道：


@ MaxMece，十分感谢你的回复，如果nan在valid.py产生是正常的，那么你生成的per_threshold是否可用（我出现了11个类别的阈值为0情况）？

**Comment by Xieshichen** [comment]
@ MaxMece你能给我一份你做Scannet Evaluation的py文件不？作者给的地址已经失效了。

**Comment by longmalongma** [comment]
> when i use the provided pre-trained model, train on area 1-6 except 5 and test on area 5, i get the admirable result and mAP is about 0.6.
> 
> but when i train on area 2-6 and test on area 1, mAP is about 0.4 and then test on area 2, mAP is about 0.1.
> 
> i try many versions of mingroupsize.txt and pergroup_thres.txt, but 0.1 is still the best result.
> 
> so i have the question about the pre-trained model, maybe i shoule retrain the pre-trained model in each cross-validation?
> 
> Thanks for your response.
> 
> ps:
> after training 200 epochs，loss is about 300-400, and grouperr: 0.02 same: 0.01 diff: 0.001 pos: 0.004



> when i use the provided pre-trained model, train on area 1-6 except 5 and test on area 5, i get the admirable result and mAP is about 0.6.
> 
> but when i train on area 2-6 and test on area 1, mAP is about 0.4 and then test on area 2, mAP is about 0.1.
> 
> i try many versions of mingroupsize.txt and pergroup_thres.txt, but 0.1 is still the best result.
> 
> so i have the question about the pre-trained model, maybe i shoule retrain the pre-trained model in each cross-validation?
> 
> Thanks for your response.
> 
> ps:
> after training 200 epochs，loss is about 300-400, and grouperr: 0.02 same: 0.01 diff: 0.001 pos: 0.004

@maxmece can you give me a py file that you did Scannet Evaluation on? The address given by the author is invalid.

**Comment by mst136** [comment]
May I ask how to obtain the AP value about s3dis？I could run generate_results.py, but it did not have this value?

**Comment by jianuo1128** [comment]
> May I ask how to obtain the AP value about s3dis？I could run generate_results.py, but it did not have this value?

@mst136 hi ,could you tell me how to prepare these txt before train? just like what mention in the this link : https://blog.csdn.net/weixin_42620141/article/details/119142860
![sgpn](https://github.com/user-attachments/assets/2c26e5ca-b7c2-473a-a994-383b7c5330a7)
